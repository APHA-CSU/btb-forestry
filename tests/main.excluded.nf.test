nextflow_process {

    name "Test Process excluded"
    script "main.nf"
    process "EXCLUDED"

    test("Should identify excluded samples") {

        when {
            params {
                today = "01Jan25"
            }
            process {
                """
                input[0] = file("${projectDir}/tests/data/Allclean_exp.csv")
                input[1] = file("${projectDir}/tests/data/highN_exp.csv")
                input[2] = file("${projectDir}/tests/data/testoutlier.txt")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
