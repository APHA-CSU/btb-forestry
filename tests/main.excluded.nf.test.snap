{
    "Should identify excluded samples": {
        "content": [
            {
                "0": [
                    "allExcluded_01Jan25.csv:md5,aa3757742432b3d0a1e7e2547d033211"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-22T14:21:51.205426255"
    }
}